### PR TITLE
Surface brightness -> solid angle conversion updates

### PR DESCRIPTION
**Pull request by gibsongreen**
### Description

This pull request is to address:

- The surface brightness drop down will become read-only text
- A new drop down will be created for the solid angle unit
- Surface brightness will update when changes to flux or the angle are made
- the “flux or sb” dropdown will no longer update automatically when making a selection to the flux/angle/sb unit drop downs

To be done:

- [x] compatibility with Specviz (flux data, surface brightness data doesn't populate (sb/flux) NEEDS UPDATE AFTER https://github.com/spacetelescope/jdaviz/pull/2931)
- [x] fix limits issue in Specviz NEEDS UPDATE AFTER https://github.com/spacetelescope/jdaviz/pull/2931
- [x] untranslatable units compatibility
- [x] pass existing tests

https://github.com/user-attachments/assets/809aff5d-6409-4a15-ac3c-326b156fce01

Currently, we enable the selection of the surface brightness unit drop down. This ticket will make the surface brightness drop down read-only. A new dropdown will be added, for selecting the denominator, the square angle unit. Surface brightness will then be calculated from the selected flux and angle unit. 

Fixes #<Issue Number>

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by rosteen — COMMENTED** [review]
I think this looks pretty good, didn't find any problems testing it. I left one suggestion, I think I can approve after that's addressed.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (gibsongreen) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:179]: Awesome! I will make the change now, I have this spot that does the same thing from my last PR so I will make the change there too:
https://github.com/spacetelescope/jdaviz/pull/2940#discussion_r1688397133

and I am going to make this same adjustment of moving the returns outside of conditionals:
https://github.com/spacetelescope/jdaviz/pull/2940#discussion_r1687124735

**Review by cshanahan1 — COMMENTED** [review]
testing this out now and is working well

it looks like app._get_display_unit('sb') is no longer available - i know you can use flux and angle to get this unit, but since other plugins are using this would it make sense to keep this way of accessing it and just make sure everything stays in sync?

**Review by rosteen — COMMENTED** [review]
One small suggestion on a comment, and we spoke offline about some variable name/conditional cleanup, so I'll hold off on approving for you to make that change. But testing in the notebook, everything works!

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py
Comment (gibsongreen) [reply jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:337]: I believe for this test, that the `flux_or_sb` traitlet is technically set to 'Surface Brightness', so by changing the `flux_unit` you aren't translating to flux you are converting sb -> sb, changing the numerator of the sb unit. Similarly changing the `angle_unit` would change the denominator.

Regarding the `app._get_display_unit('sb')`, I do see that this isn't working anymore. There is a GlobalDisplayUnitChanged message that gets broadcasted whenever a change to the y_unit is made. I will make a change to ensure that you can still access the display unit this way.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (gibsongreen) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py]: Good catch!

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (gibsongreen) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py]: This has now been removed

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py
Comment (cshanahan1) [reply jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:337]: i see what you mean about the tests, that seems fine then i redact my comment

**Review by rosteen — APPROVED** [review]
I think this looks good now - there are a couple things I noticed that I think were from the previous PR, not changed here, so I might open a quick follow up after this is merged. But no need to hold this up since this PR didn't introduce them.

**Comment by rosteen** [comment]
> it looks like app._get_display_unit('sb') is no longer available

It doesn't look like this PR changed app.py, any idea when this changed?

**Comment by cshanahan1** [comment]
> > it looks like app._get_display_unit('sb') is no longer available
> 
> @cshanahan1 It doesn't look like this PR changed app.py, any idea when this changed?

i think its because its trying to access the 'sb_unit_selected' from the unit conversion plugin, which now fails

**Comment by gibsongreen** [comment]
> > > it looks like app._get_display_unit('sb') is no longer available
> > 
> > 
> > @cshanahan1 It doesn't look like this PR changed app.py, any idea when this changed?
> 
> i think its because its trying to access the 'sb_unit_selected' from the unit conversion plugin, which now fails

It comes from removing `sb_unit_selected`, below is what gets called when you do this 

`app._get_display_unit('sb')`

       return getattr(self._jdaviz_helper.plugins.get('Unit Conversion')._obj,
                           f'{axis}_unit_selected')

In unit_conversion.py I can change the traitlet from self.sb_unit (new to this PR) to self.sb_unit_selected and it will function as it did in the past, any opposition to this change?

**Comment by rosteen** [comment]
Ahh I see, good call - I didn't read to the end of the method 😅. I agree it might be worth adding a manual case for `sb` in `get_angle_unit` that combines the results for `flux_unit` and `angle_unit` automatically.

**Comment by rosteen** [comment]
> In unit_conversion.py I can change the traitlet from self.sb_unit (new to this PR) to self.sb_unit_selected and it will function as it did in the past

This sounds simpler than what I just suggested.

**Comment by gibsongreen** [comment]
> > In unit_conversion.py I can change the traitlet from self.sb_unit (new to this PR) to self.sb_unit_selected and it will function as it did in the past
> 
> This sounds simpler than what I just suggested.

@cshanahan1  let me know if this suffices, or if the other suggested change is better for the generalization of units ticket I candy that instead

**Comment by rosteen** [comment]
That change works for me, my approval stands 🙂

**Review by cshanahan1 — APPROVED** [review]
yep that works for me now, thanks!

**Pull request merged**
